sdktests/server_side_events_summary.go

```diff
@@ -51,40 +51,35 @@ func doServerSideSummaryEventBasicTest(t *ldtest.T) {
 	client := NewSDKClient(t, dataSource, events)
 
 	// evaluations for flag1: two for userA producing value1a, one for userB producing value1b
-	_ = client.EvaluateFlag(t, servicedef.EvaluateFlagParams{FlagKey: flag1.Key, User: &userA, DefaultValue: default1})
-	_ = client.EvaluateFlag(t, servicedef.EvaluateFlagParams{FlagKey: flag1.Key, User: &userB, DefaultValue: default1})
-	_ = client.EvaluateFlag(t, servicedef.EvaluateFlagParams{FlagKey: flag1.Key, User: &userA, DefaultValue: default1})
+	_ = client.EvaluateFlag(t, servicedef.EvaluateFlagParams{FlagKey: flag1.Key, User: userA, DefaultValue: default1})
+	_ = client.EvaluateFlag(t, servicedef.EvaluateFlagParams{FlagKey: flag1.Key, User: userB, DefaultValue: default1})
+	_ = client.EvaluateFlag(t, servicedef.EvaluateFlagParams{FlagKey: flag1.Key, User: userA, DefaultValue: default1})
 
 	// evaluations for flag2: one for userA producing value2a
-	_ = client.EvaluateFlag(t, servicedef.EvaluateFlagParams{FlagKey: flag2.Key, User: &userA, DefaultValue: default2})
+	_ = client.EvaluateFlag(t, servicedef.EvaluateFlagParams{FlagKey: flag2.Key, User: userA, DefaultValue: default2})
 
 	client.FlushEvents(t)
 	payload := events.ExpectAnalyticsEvents(t, defaultEventTimeout)
 
 	m.In(t).Assert(payload, m.Items(
-		EventIsIndexEvent(mockld.SimpleEventUser(userA)),
-		EventIsIndexEvent(mockld.SimpleEventUser(userB)),
-		m.JSONMap().Should(m.MapOf(
-			m.KV("kind", m.Equal("summary")),
-			m.KV("startDate", ValueIsPositiveNonZeroInteger()),
-			m.KV("endDate", ValueIsPositiveNonZeroInteger()),
-			m.KV("features", m.MapOf(
-				m.KV(flag1.Key, m.MapOf(
-					m.KV("default", m.JSONEqual(default1)),
-					m.KV("counters", m.ItemsInAnyOrder(
-						flagCounter("value1a", 0, flag1.Version, 2),
-						flagCounter("value1b", 1, flag1.Version, 1),
-					)),
+		IsIndexEvent(),
+		IsIndexEvent(),
+		IsValidSummaryEventWithFlags(
+			m.KV(flag1.Key, m.MapOf(
```

Comment:
Here I've continued to use the `MapOf` method, which I stopped using in most other cases, because the summary event data structure is one big nested thing, and all of the properties within `features` are predictable and relevant to this test.